packages/api/src/routes/medical/patient.ts

```diff
@@ -478,4 +484,34 @@ router.get(
   })
 );
 
+/** ---------------------------------------------------------------------------
+ * POST /patient/search
+ *
+ * Searches for a patient based on a demographic paylaod and returns the matched patient if exists.
+ * If no patient exists, return 404.
+ *
+ * @return The matched patient.
+ */
+router.post(
+  "/search",
```

Comment:
There was an explicit customer request in the issue to *not* return more than one patient, and instead throw an error. Allowing filtering along the `GET /patient` wouldn't naturally have this behavior (i.e. it should always simply return any matched patients, or an empty list).

I can ignore this comment and migrate to the `GET /patient` approach if we want.

CAVEAT: `getPatientByDemo` function always returns a single patient via a first matching function. This affects the `POST /patient` behavior and the new search behavior. To really follow the user comment, we should undo this and through a 404 when no patient is matched OR more than one patient is matched (or some other error for the 2+ match case). But, we'd need to decide how to handle the `POST /patient` case (either return more than one patient or move the first matching logic into the handler for this endpoint.

Reply:
Taken to [messaging](https://metriport.slack.com/archives/C04DMKE9DME/p1715630069653729)

Reply:
Plan:
1. switch endpoint language to `match` instead of `search`
2. add ordering to PatientLoader to ensure getPatientByDemo returns the earliest created matching patient
3. log a warning if matchPatients returns more than one user within getPatientByDemo